The setup.py creates a compiler object to check for the `clock_gettime`
function. However, when it does so, it uses the default settings, i.e., `cc`.
When the compiler object gets created as a part of the normal build, the
settings get customized to point to the sysconfig data. This is where it would
normally update to our compilers and flags. So, we need to call this method to
ensure that the function check is using the conda compiler and sysroot.

diff --git a/setup.py b/setup.py
index 95a3cb8..3bf9b62 100644
--- a/setup.py
+++ b/setup.py
@@ -103,6 +103,8 @@ if not WINDOWS:
     # check for clock_gettime, link with librt for glibc<2.17
     from dev import ccompiler
     compiler = ccompiler.new_compiler()
+    from distutils.sysconfig import customize_compiler
+    customize_compiler(compiler)
     if not compiler.has_function('clock_gettime(0,NULL)', includes=['time.h']):
         if compiler.has_function('clock_gettime(0,NULL)', includes=['time.h'], libraries=['rt']):
             libraries.append('rt')
